shoop/themes/classic_gray/templates/classic_gray/shoop/front/basket/partials/_basket_content.jinja

```diff
@@ -32,24 +32,27 @@
                     {% if product and product.is_package_parent() %}
                         {{ macros.render_package_children(product) }}
                     {% endif %}
-                    <div class="quantity">
-                        {% if product and line.can_change_quantity %}
-                        <div class="input-group">
-                            <div class="input-group-addon">
-                                {% trans %}Qty{% endtrans %}
+
+                    {% if not line.line_id.startswith("free_") %}
+                        <div class="quantity">
+                            {% if product and line.can_change_quantity %}
```

Comment:
Put your "free_" checks in line.can_change_quantity. Not sure if this free_ thing is ok either.